editor/editor_node.cpp

```diff
@@ -2741,47 +2741,42 @@ Dictionary EditorNode::_get_main_scene_state() {
 	state["property_edit_offset"] = get_property_editor()->get_scene_tree()->get_vscroll_bar()->get_value();
 	state["saved_version"] = saved_version;
 	state["node_filter"] = scene_tree_dock->get_filter();
+	return state;
+}
+
+void EditorNode::_set_main_scene_state(Dictionary p_state, Node *p_for_scene) {
+
 	int current = -1;
+	int index = p_state["editor_index"];
+	int n2d = 0, n3d = 0;
```

Comment:
It would also be clearer if those were defined where they used to be, no other code chunk needs them so they don't need to be in the main scope. Remember that this is C++, not C, you don't have to (and shouldn't, unless relevant) declare local variables upfront in a method.